source/references/glossary.markdown

```diff
@@ -84,166 +129,228 @@ See [external node classifier](#external-node-classifier).
+
+In a standard Puppet client-server deployment, the server is known as the **Puppet master**. The Puppet master [compiles](#compilation) and serves configuration [catalogs](#catalog) on demand to Puppet [agents](#agent) on client [nodes](#node).
 
-The puppet master uses an HTTP server to provide catalogs. It can run as a standalone daemon process with a built-in web server, or it can be managed by a production-grade web server that supports the rack API. The built-in web server is meant for testing, and is not suitable for use with more than ten nodes.
+The Puppet master provides catalogs to agents using an HTTP server. It can run as a standalone daemon with a built-in web server, or as part of [Puppet Server](#puppet-server). (We don't recommend using the built-in daemon when managing more than 10 nodes.)
+
+For more information, see the [Overview of Puppet's Architecture](/puppet/latest/reference/architecture.html) page in the Puppet documentation.
+
+### masterless
+
+"**Masterless**" is a term often used to refer to a Puppet [node](#node) that does not rely on a [master](#master) for its [catalog](#catalog). See [agent](#agent).
 
 ### manifest
 
-A file containing code written in the Puppet language, and named with the `.pp` file extension. The Puppet code in a manifest can:
+(or **Puppet code**)
+
+A **manifest** file contains code written in the [Puppet language](#puppet-language) and is named with the `.pp` file extension. The **Puppet code** in a manifest can:
 
 * [Declare](#declare) [resources](#resource) and [classes](#class)
 * Set [variables](#variable)
 * Evaluate [functions](#function)
-* [Define](#define) [classes](#class), [defined types](#type-defined), and [nodes](#node-definition)
+* [Define](#define) [classes](#class), [defined types](#type-defined), [functions](#function), and [nodes](#node-definition)
 
+Most manifests are contained in [modules](#module). Every manifest in a module should [define](#define) a single class, defined type, or function.
 
-Most manifests are contained in [modules](#module). Every manifest in a module should [define](#define) a single [class](#class) or [defined type](#type-defined).
+The Puppet [master](#master) service reads an [environment's](#environment) [main manifest](#main-manifest). This manifest usually defines [nodes](#node-definition), so that each managed [agent](#agent) receives a unique [catalog](#catalog).
 
-The puppet master service reads a single "site manifest," usually located at `/etc/puppet/manifests/site.pp`. This manifest usually defines [nodes](#node-definition), so that each managed [agent node](#node) will receive a unique catalog.
+See also [main manifest](#main-manifest).
+
+### manifest ordering
+
+Puppet uses **manifest ordering** to apply [resources](#resource) in the order that they're declared in their [manifests](#manifest). For more information, see the [Relationships and Ordering][] page in the [Puppet language](#puppet-language) reference.
```

Comment:
I think it's worth saying that this is for unrelated resources, and that related resources should have explicit relationships.